### PR TITLE
Mark spinejson resource type as textual

### PR DESCRIPTION
**Pull request by vlaaad**
This change will enable Search in Files in spinejson files after https://github.com/defold/defold/pull/7642 is merged.

Related to https://github.com/defold/defold/issues/7586

### CODE REVIEW

File: defold-spine/editor/src/spineext.clj [hunk 1]
```diff
@@ -1008,6 +1008,7 @@
    (workspace/register-resource-type workspace
                                      :ext spine-json-ext
                                      :node-type SpineSceneJson
+                                     :textual? true
```
Comment (JCash) [inline defold-spine/editor/src/spineext.clj:1011]: While it _is_ textual, I do wonder if it in this case reflects our intended reason for "Search in files".

E.g.this is a 3'rd party format, which it's not always necessary to dig into (depending on how ppl use the search-in-files).
It is not likely they'll change anything in there, and finding strings in a spine json is probably not very common.

In short, does it make sense to have another flag (e.g. "proprietary") and also have an extra checkbox in the search dialog for searching in external files as well?

I mean, same goes for binary files as well. I search though files regardless of binary/text, and I think it's reasonable that the users might was some granularity as well.

Just a thought.

Comment (matgis) [reply defold-spine/editor/src/spineext.clj:1011]: Couldn't you want to search for bone names or things like that inside these files. I was thinking you might refer to these names from script files for example, and might want to ensure you've renamed every occurrence in both scripts and spine scenes.

**Review by matgis — APPROVED** [review]

File: defold-spine/editor/src/spineext.clj (hunk 1 above)
Comment (JCash) [reply defold-spine/editor/src/spineext.clj:1011]: Sure, but that goes for binary files as well then. (case in point: .glb)

As for searching those things, wouldn't it be relevant to search our _parsed_ data then? E.g. which bone names did we find in the file?

Comment (matgis) [reply defold-spine/editor/src/spineext.clj:1011]: Yep! And this might in fact happen soon. :)

**Pull request merged**